AS-3691 - Updated objectsToDocument function to deal with array syntax

JamesXiatech opened the pull request:
Refactored the objectsToDocument function to deal with array syntax in dot notation such as EventMetadata.Times[1].UTC

Commit:
Updated objectsToDocument function to deal with array syntax

tbal999 commented on jlib/new.go:
```diff
@@ -259,23 +319,32 @@ func ObjectsToDocument(input interface{}) (interface{}, error) {
+
+		// Try "Val" first, then fall back to "Value"
+		var value interface{}
+		if val, exists := item["Val"]; exists {
```
is this just a backward compatibility thing? I.e some devs use "Val" and others use "Value"?

xiatech-ricky replied:
Assume this to be the case. Now outdated as also added nil check

tbal999 commented on jlib/new.go:
```diff
@@ -259,23 +319,32 @@ func ObjectsToDocument(input interface{}) (interface{}, error) {
-		// Call setValue for each item to set the value in the output map
+
 		code, ok := item["Code"].(string)
 		if !ok {
```
should this return an error? If it's missing code, could be a user error.
Or is that expected behaviour, if yes, you can ignore this comment.

xiatech-ricky replied:
@JamesXiatech - I'm not sure on the behaviour, could you confirm if missing `Code` is good to disregard this line and drop it, or if you need it to throw an error so it can be correct?

xiatech-ricky replied:
Confirmed with @JamesXiatech that the desired behaviour is to error. Code has been updated to replace the `continue` with an Error.

Commit:
chore: extended testing chassis

Commit:
chore: switch to goccy

Commit:
chore: some comments

Commit:
chore: simplify chassis

Commit:
chore: clean up readme

xiatech-ricky committed:
chore(AS-3797): add test coverage

fix nil pointer issue, add unit tests, add data driven tests, fix bug in data driven framework

xiatech-ricky committed:
chore(as-3797): throw error if no Code found

Changed `continue` to throw an Error if no Code is found in the array item list for $objectsToDocument
